### PR TITLE
Migrate to mqtt-go

### PR DESCRIPTION
**Pull request by at-wat**

### CODE REVIEW

**Comment by at-wat** [comment]
@t-asaka @kt3k @kamatama41 please take a look

File: device.go [hunk 1]
```diff
@@ -12,185 +12,5 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/*
-Package awsiotdev implements offline queueing and reconnecting features of MQTT protocol.
-*/
+// Package awsiotdev implements offline queueing and reconnecting features of MQTT protocol.
 package awsiotdev
```
Comment (kt3k) [inline device.go:16]: Do we still need this file?

Comment (at-wat) [reply device.go:16]: This is just for GoDoc. `awsiotdev.go` may be better filename.

Comment (at-wat) [reply device.go]: updated the filename

File: shadow/state.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+			}
+		default:
+			state[key] = v
+		}
+	}
+	return nil
+}
+
+func hasUpdate(s json.RawMessage) bool {
+	switch {
```
Comment (kamatama41) [inline shadow/state.go:116]: `return len(s) != 0` seems simpler.

**Comment by at-wat** [comment]
Split thing shadow implementation to #70

File: shadow/state.go (hunk 2 above)
Comment (at-wat) [reply shadow/state.go]: fixed at https://github.com/seqsense/aws-iot-device-sdk-go/pull/70/commits/fbd66c060d7029314de87e83ad7f00359f24f467

**Pull request merged**